src/nbody_boundary_conditions.jl

```diff
@@ -29,4 +29,48 @@ InfiniteBox() = InfiniteBox(SVector(-Inf, Inf, -Inf, Inf, -Inf, Inf))
 
 struct CubicPeriodicBoundaryConditions{cType <: Real} <: BoundaryConditions
     L::cType
+end
+
+function apply_boundary_conditions!(ri, rj, pbc::PeriodicBoundaryConditions, R2)
+    rij = @MVector [Inf, Inf, Inf]
+    success = false
+    rij2 = 0
+
+    for dx in [0,-pbc[2],pbc[2]], dy in [0,-pbc[4],pbc[4]], dz in [0,-pbc[6],pbc[6]]
+        rij = @MVector [ri[1] - rj[1] + dx, ri[2] - rj[2] + dy, ri[3] - rj[3] + dz]
```

Comment:
This is a good use. Note that this allocates on v0.6 but it's optimized away on v0.7

Reply:
Got it